src/platform/datapath_raw_xdp_win.c

```diff
@@ -1450,52 +1450,58 @@ CxPlatDpRawPlumbRulesOnSocket(
-            Rules[2].Redirect.Target = NULL;
-
-            Rules[3].Match = XDP_MATCH_TCP_QUIC_FLOW_DST_CID;
-            Rules[3].Pattern.QuicFlow.UdpPort = Socket->LocalAddress.Ipv4.sin_port;
-            Rules[3].Pattern.QuicFlow.CidLength = Socket->CibirIdLength;
-            Rules[3].Pattern.QuicFlow.CidOffset = Socket->CibirIdOffsetDst;
-            Rules[3].Action = XDP_PROGRAM_ACTION_REDIRECT;
-            Rules[3].Redirect.TargetType = XDP_REDIRECT_TARGET_TYPE_XSK;
-            Rules[3].Redirect.Target = NULL;
-
+            if (Socket->ReserveAuxTcpSock) {
+                Rules[2].Match = XDP_MATCH_TCP_QUIC_FLOW_SRC_CID;
+                Rules[2].Pattern.QuicFlow.UdpPort = Socket->LocalAddress.Ipv4.sin_port;
+                Rules[2].Pattern.QuicFlow.CidLength = Socket->CibirIdLength;
+                Rules[2].Pattern.QuicFlow.CidOffset = Socket->CibirIdOffsetSrc;
+                Rules[2].Action = XDP_PROGRAM_ACTION_REDIRECT;
+                Rules[2].Redirect.TargetType = XDP_REDIRECT_TARGET_TYPE_XSK;
+                Rules[2].Redirect.Target = NULL;
+
+                Rules[3].Match = XDP_MATCH_TCP_QUIC_FLOW_DST_CID;
+                Rules[3].Pattern.QuicFlow.UdpPort = Socket->LocalAddress.Ipv4.sin_port;
+                Rules[3].Pattern.QuicFlow.CidLength = Socket->CibirIdLength;
+                Rules[3].Pattern.QuicFlow.CidOffset = Socket->CibirIdOffsetDst;
+                Rules[3].Action = XDP_PROGRAM_ACTION_REDIRECT;
+                Rules[3].Redirect.TargetType = XDP_REDIRECT_TARGET_TYPE_XSK;
+                Rules[3].Redirect.Target = NULL;
+            }
 
             memcpy(Rules[0].Pattern.QuicFlow.CidData, Socket->CibirId, Socket->CibirIdLength);
             memcpy(Rules[1].Pattern.QuicFlow.CidData, Socket->CibirId, Socket->CibirIdLength);
-            memcpy(Rules[2].Pattern.QuicFlow.CidData, Socket->CibirId, Socket->CibirIdLength);
-            memcpy(Rules[3].Pattern.QuicFlow.CidData, Socket->CibirId, Socket->CibirIdLength);
-
-            Rules[4].Match = XDP_MATCH_TCP_CONTROL_DST;
-            Rules[4].Pattern.Port = Socket->LocalAddress.Ipv4.sin_port;
-            Rules[4].Action = XDP_PROGRAM_ACTION_REDIRECT;
-            Rules[4].Redirect.TargetType = XDP_REDIRECT_TARGET_TYPE_XSK;
-            Rules[4].Redirect.Target = NULL;
-            RulesSize = 5;
 
+            if (Socket->ReserveAuxTcpSock) {
+                memcpy(Rules[2].Pattern.QuicFlow.CidData, Socket->CibirId, Socket->CibirIdLength);
+                memcpy(Rules[3].Pattern.QuicFlow.CidData, Socket->CibirId, Socket->CibirIdLength);
+                Rules[4].Match = XDP_MATCH_TCP_CONTROL_DST;
+                Rules[4].Pattern.Port = Socket->LocalAddress.Ipv4.sin_port;
+                Rules[4].Action = XDP_PROGRAM_ACTION_REDIRECT;
+                Rules[4].Redirect.TargetType = XDP_REDIRECT_TARGET_TYPE_XSK;
+                Rules[4].Redirect.Target = NULL;
+                RulesSize = 5;
+            } else {
+                RulesSize = 2;
```

Comment:
nit: Maybe a personal taste, but I would put the size to 2 right below the rules 0 and 1 are set, and overwrite it to 5 in the if block.

It reduces the risk the count ends up out of sync in the future